File: classes/parte-6-datos-y-estructuras/103-propiedad-y-ciclo-de-vida-de-los-datos/implementaciones/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	line, _ := bufio.NewReader(os.Stdin).ReadString('\n')
	n, _ := strconv.Atoi(strings.TrimSpace(line))
	valor := 0
	func() {
		defer func() { /* se libera al salir */ }()
		valor = n
	}()
	fmt.Printf("valor=%d estado=liberado\n", valor)
}

File: classes/parte-6-datos-y-estructuras/103-propiedad-y-ciclo-de-vida-de-los-datos/implementaciones/python/main.py
import sys


class Recurso:
    def __init__(self, valor):
        self.valor = valor

    def __enter__(self):
        return self

    def __exit__(self, *args):
        pass  # aquí se liberaría


n = int(sys.stdin.readline())
with Recurso(n) as r:
    valor = r.valor
print(f"valor={valor} estado=liberado")
